### ISSUE TITLE
Add a adapter for Async Storage and Encrypted Storage (React Native)

### ISSUE THREAD

**Issue by rushelex** [yumauri/effector-storage#30]
Hi.
I wrote a simple driver for Async Storage and Encrypted Storage for a React Native application.
I can see from the source code that the driver needs to be converted to a persist function, but I'm not sure I can get it right to match the current styleguide.
Can I just submit a PR with my written withAsyncStorageAdapter and withEncryptedStorageAdapter functions?

Example:

```typescript
// asyncStorageAdapter.ts

import { Domain } from "effector";
import { persist, StorageAdapter } from "effector-storage";
import AsyncStorage from "@react-native-async-storage/async-storage";

const adapter: StorageAdapter = (key) => ({
  get: async () => {
    return AsyncStorage.getItem(key).then((value) =>
      value ? JSON.parse(value) : undefined,
    );
  },
  set: async (value) => {
    return AsyncStorage.setItem(key, JSON.stringify(value));
  },
});

export const withAsyncStorageAdapter = (domain: Domain) => {
  domain.onCreateStore((store) => persist({ store, adapter }));
};
```

```typescript
// encryptedStorageAdapter.ts

import { Domain } from "effector";
import { persist, StorageAdapter } from "effector-storage";
import EncryptedStorage from "react-native-encrypted-storage";

const adapter: StorageAdapter = (key) => ({
  get: async () => {
    return EncryptedStorage.getItem(key).then((value) =>
      value ? JSON.parse(value) : undefined,
    );
  },
  set: async (value) => {
    return EncryptedStorage.setItem(key, JSON.stringify(value));
  },
});

export const withEncryptedStorageAdapter = (domain: Domain) => {
  domain.onCreateStore((store) => persist({ store, adapter }));
};
```

```typescript
// model.ts

const domain = createDomain('my example domain');
withAsyncStorageAdapter(domain);
```

**Comment by yumauri** [comment]
I see those two adapters quite similar, does those storages have same API? Maybe it will be more convenient to add single configurable async adapter?

**Comment by rushelex** [comment]
> I see those two adapters quite similar, does those storages have same API? Maybe it will be more convenient to add single configurable async adapter?

It will be clear from the previous answer that the basic getItem() and setItem() methods are similar, so if you leave only these methods and the clear() method (if needed), you can combine these two adapters into one configurable one.
For example:

```typescript
import { Domain } from "effector";
import { persist, StorageAdapter } from "effector-storage";
import AsyncStorage from "@react-native-async-storage/async-storage";
import EncryptedStorage from "react-native-encrypted-storage";

interface AdapterOptions {
  storage: typeof AsyncStorage | typeof EncryptedStorage;
}

const defaultAdapterOptions: AdapterOptions = {
  storage: AsyncStorage,
};

const adapter = (options: AdapterOptions): StorageAdapter => {
  return (key) => ({
    get: async () => {
      return options.storage
        .getItem(key)
        .then((value) => (value ? JSON.parse(value) : undefined));
    },
    set: async (value) => {
      return options.storage.setItem(key, JSON.stringify(value));
    },
  });
};

export const withEncryptedStorageAdapter = (
  domain: Domain,
  options: AdapterOptions = defaultAdapterOptions,
) => {
  domain.onCreateStore((store) => {
    return persist({ store, adapter: adapter(options) });
  });
};
```

This construction, I think, can easily be developed in the future by adding more options, etc.

**Comment by rushelex** [comment]
@yumauri Yes, I created a PR #31

**Comment by yumauri** [comment]
Thank you!
I'll leave this issue open until new release is landed.

**Comment by yumauri** [comment]
Published new [release 4.5.0](https://github.com/yumauri/effector-storage/releases/tag/v4.5.0)

@rushelex try, please, in your project:

```typescript
// asyncStorageAdapter.ts

import type { Domain } from "effector";
import { persist } from 'effector-storage/rn/async';

export const withAsyncStorageAdapter = (domain: Domain) => {
  domain.onCreateStore((store) => persist({ store }));
};
```

```typescript
// encryptedStorageAdapter.ts

import type { Domain } from "effector";
import { persist } from 'effector-storage/rn/encrypted';

export const withEncryptedStorageAdapter = (domain: Domain) => {
  domain.onCreateStore((store) => persist({ store }));
};
```

⚠️ Note, that you still have to install AsyncStorage and EncryptedStorage, `effector-storage` doesn't have them as dependencies, nor peer dependencies, and **will not** install them automatically!

**Comment by yumauri** [comment]
Closing this, @rushelex if you have any issues with those new adapters — feel free to reopen!
